nx/lib/nx/shape.ex

```diff
@@ -1721,24 +1721,36 @@ defmodule Nx.Shape do
       iex> Nx.Shape.cholesky({4, 4}, [:x, :y])
       {{4, 4}, [:x, :y]}
 
+      iex> Nx.Shape.cholesky({3, 3, 3}, [:x, :y, :z])
+      {{3, 3, 3}, [:x, :y, :z]}
+
   ## Error Cases
 
-      iex> Nx.Shape.cholesky({3, 2}, [:x, :y])
-      ** (ArgumentError) tensor must be a square matrix, got shape: {3, 2}
+      iex> Nx.Shape.cholesky({2, 3, 2}, [:x, :y, :z])
+      ** (ArgumentError) tensor must be a batch of square matrices, got shape: {2, 3, 2}
 
-      iex> Nx.Shape.cholesky({3, 3, 3}, [:x, :y, :z])
-      ** (ArgumentError) tensor must have rank 2, got rank 3 with shape {3, 3, 3}
+      iex> Nx.Shape.cholesky({3}, [:x])
+      ** (ArgumentError) tensor must have at least rank 2, got rank 1 with shape {3}
   """
-  def cholesky({n, n}, names), do: {{n, n}, names}
+  def cholesky(shape, names) when tuple_size(shape) > 1 do
+    rank = tuple_size(shape)
+    matrix_shape = {elem(shape, rank - 2), elem(shape, rank - 1)}
+
+    unless match?({n, n}, matrix_shape) do
+      raise(
+        ArgumentError,
+        "tensor must be a batch of square matrices, got shape: #{inspect(shape)}"
```

Comment:
I'm not sure if we should mention batches in the case where the input is a plain matrix, for all batch-related error messages

Reply:
I changed it to `must be a square matrix or a batch of square matrices`, but if you come up with a better message I'll be happy to change it

Reply:
This is a bit verbose, but it's clear on what's expect, so I think it's fine!